FileManager Toolbar customization documentation

ntacheva opened the pull request.

dimodi commented on components/filemanager/toolbar.md:
```diff
@@ -10,36 +10,403 @@ position: 15
 # FileManager Toolbar
 
-The Toolbar of the FileManager provides the following built-in commands:
+The FileManager includes a built-in Toolbar that contains several commands. You may use it as is or customize it to include your desired tools.
```
Try to provide some more value in this introductory sentence, otherwise the reader may as well skip it.

```suggestion
The FileManager includes a built-in Toolbar that contains commands for adding new files and folders, sorting and changing the view, and more. You can use the toolbar as is or customize it to include the desired built-in or custom tools.
```

dimodi commented on components/filemanager/toolbar.md:
```diff
@@ -10,36 +10,403 @@ position: 15
+>caption In this article:
+* [Default Toolbar](#default-toolbar)
+* [Custom Toolbar](#custom-toolbar)
+    * [Example](#example)
```
Is the heading for custom tools missing on purpose?

dimodi commented on components/filemanager/toolbar.md:
```diff
@@ -10,36 +10,403 @@ position: 15
+## Default Toolbar
 
-The `New Folder` button allows creating of a new folder in the current location. Read how to handle the creation in the [FileManager - Events]({%slug filemanager-events%}).
+The default Toolbar will be automatically rendered when you declare the FileManager component. If you want to use the default version, you do not need additional Toolbar configuration.
```
```suggestion
The default Toolbar renders automatically when you declare the FileManager component. Additional Toolbar configuration is not required.
```

dimodi commented on components/filemanager/toolbar.md:
```diff
@@ -10,36 +10,403 @@ position: 15
+The default Toolbar will be automatically rendered when you declare the FileManager component. If you want to use the default version, you do not need additional Toolbar configuration.
 
-## Upload
+The default Toolbar contains the following built-in tools (rendered in this exact order):
```
For SEO and AI purposes.

```suggestion
The default FileManager Toolbar contains the following built-in tools (rendered in this exact order):
```

dimodi commented on components/filemanager/toolbar.md:
```diff
@@ -10,36 +10,403 @@ position: 15
-## Sort
+| Tool | Description |
+| --- | --- |
+| New Folder| A button that allows creating of a new folder in the current location. Read how to handle the creation in the [FileManager - Events]({%slug filemanager-events%}). |
```
```suggestion
| New Folder | A button that creates a new folder in the current location. Read how to handle the creation at [FileManager Events]({%slug filemanager-events%}). |
```

dimodi commented on components/filemanager/toolbar.md:
```diff
@@ -10,36 +10,403 @@ position: 15
+| --- | --- |
+| New Folder| A button that allows creating of a new folder in the current location. Read how to handle the creation in the [FileManager - Events]({%slug filemanager-events%}). |
+| Upload | A button that allows updloading of new files. Clicking the button will open a dialog with integrated [Upload component]({%slug upload-overview%}). Read more in the [FileManager - Upload]({%slug filemanager-upload%}) article. |
+| Sort Direction | A ButtonGroup with ToggleButtons that allows selecting the sort direction(ascending or descending) to sort the files in the current location. Read more in the [FileManager - Sort]({%slug filemanager-sort%}) article. |
```
```suggestion
| Sort Direction | A ButtonGroup with ToggleButtons that allows selecting the sort direction(ascending or descending) to sort the files in the current location. Read more in the [FileManager Sort]({%slug filemanager-sort%}) article. |
```

dimodi commented on components/filemanager/toolbar.md:
```diff
@@ -10,36 +10,403 @@ position: 15
+| New Folder| A button that allows creating of a new folder in the current location. Read how to handle the creation in the [FileManager - Events]({%slug filemanager-events%}). |
+| Upload | A button that allows updloading of new files. Clicking the button will open a dialog with integrated [Upload component]({%slug upload-overview%}). Read more in the [FileManager - Upload]({%slug filemanager-upload%}) article. |
+| Sort Direction | A ButtonGroup with ToggleButtons that allows selecting the sort direction(ascending or descending) to sort the files in the current location. Read more in the [FileManager - Sort]({%slug filemanager-sort%}) article. |
+| Sort By | A SplitButton that provides an option to select the desired sort member to sort by. Read more in the [FileManager - Sort]({%slug filemanager-sort%}) article. |
```
```suggestion
| Sort By | A SplitButton that provides an option to select the desired sort member to sort by. Read more in the [FileManager Sort]({%slug filemanager-sort%}) article. |
```

dimodi commented on components/filemanager/toolbar.md:
```diff
@@ -10,36 +10,403 @@ position: 15
+| Upload | A button that allows updloading of new files. Clicking the button will open a dialog with integrated [Upload component]({%slug upload-overview%}). Read more in the [FileManager - Upload]({%slug filemanager-upload%}) article. |
+| Sort Direction | A ButtonGroup with ToggleButtons that allows selecting the sort direction(ascending or descending) to sort the files in the current location. Read more in the [FileManager - Sort]({%slug filemanager-sort%}) article. |
+| Sort By | A SplitButton that provides an option to select the desired sort member to sort by. Read more in the [FileManager - Sort]({%slug filemanager-sort%}) article. |
+| Views | A ButtonGroup with ToggleButtons that allows you to control how the content of the selected directory will be visualized. The users can choose between two options. Read more in the [FileManager - Views]({%slug filemanager-views%}) article. |
```
```suggestion
| Views | A ButtonGroup with ToggleButtons that allows you to control how the content of the selected directory will be visualized. The users can choose between two options. Read more in the [FileManager Views]({%slug filemanager-views%}) article. |
```

dimodi commented on components/filemanager/toolbar.md:
```diff
@@ -10,36 +10,403 @@ position: 15
+| Sort Direction | A ButtonGroup with ToggleButtons that allows selecting the sort direction(ascending or descending) to sort the files in the current location. Read more in the [FileManager - Sort]({%slug filemanager-sort%}) article. |
+| Sort By | A SplitButton that provides an option to select the desired sort member to sort by. Read more in the [FileManager - Sort]({%slug filemanager-sort%}) article. |
+| Views | A ButtonGroup with ToggleButtons that allows you to control how the content of the selected directory will be visualized. The users can choose between two options. Read more in the [FileManager - Views]({%slug filemanager-views%}) article. |
+| View Details | A Switch that toggles the visibility of the [Preview Pane]({%slug filemanager-preview-pane%}). By default its value is `Off` and you can enable it to see the selected file details. |
```
SEO

```suggestion
| View Details | A Switch that toggles the visibility of the [FileManager Preview Pane]({%slug filemanager-preview-pane%}). By default the preview pane is hidden. |
```

dimodi commented on components/filemanager/toolbar.md:
```diff
@@ -10,36 +10,403 @@ position: 15
-The FileManager allows searching files in the current location via a Searchbar in the Toolbar. Read more in the [FileManager - Search]({%slug filemanager-search%}) article.
+@[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)
+
+| Tool Component | Description |
```
Optionally, you can consider a single table for the built-in tools and tool tags. This will save space and avoid repetition:
https://docs.telerik.com/blazor-ui/components/pdfviewer/toolbar

dimodi commented on components/filemanager/toolbar.md:
`````diff
@@ -10,36 +10,403 @@ position: 15
+</TelerikFileManager>
+````
+
+### Example
`````
Using JavaScript for a built-in feature is a but unusual. Can we think of something simpler for our customers? For example:
* The custom tool can perform an action that is normally in the context menu (delete).
* The custom tool can display the value of a custom property in the data item.

dimodi commented on components/filemanager/toolbar.md:
```diff
@@ -10,36 +10,403 @@ position: 15
+            gridDesign
+        };
+
+        return await Task.FromResult(files);
```
This syntax is unnecessarily complex. Probably that's a copy-paste, so if there is no time, we can refactor everywhere another time.

yordan-mitev commented on components/filemanager/toolbar.md:
```diff
@@ -10,36 +10,403 @@ position: 15
+| `FileManagerToolBarViewDetailsTool` | Renders a Switch to toggle the Preview Pane for the selected file. |
+| `FileManagerToolBarSearchTool` | Renders a search box. | 
+
+### Custom tools
```
```suggestion
### Custom Tools
```

yordan-mitev commented on components/filemanager/toolbar.md:
```diff
@@ -10,36 +10,403 @@ position: 15
+
+### Custom tools
+
+To create a custom tool, declare a `<FileManagerToolBarCustomTool>` inside `<FileManagerToolBar>` tag. Place the content of your tool inside the `<FileManagerToolBarCustomTool>` tag.
```
```suggestion
To create a custom tool, declare a `<FileManagerToolBarCustomTool>` inside the `<FileManagerToolBar>` tag. Place the content of your tool inside the `<FileManagerToolBarCustomTool>` tag.
```

yordan-mitev commented on components/filemanager/toolbar.md:
```diff
@@ -10,36 +10,403 @@ position: 15
+
+    private async Task OnCreateHandler(FileManagerCreateEventArgs args)
+    {
+        // the new item data is hardcoded for the purpose of the example
```
* Format each complete sentence as such: start it with a capital letter and end it with the corresponding punctuation.
* Use "hard-code" instead of "hardcode".
```suggestion
        // The new item data is hard-coded for the purpose of the example.
```

yordan-mitev commented on components/filemanager/toolbar.md:
`````diff
@@ -10,36 +10,403 @@ position: 15
+
+The following example demonstrates the customization of the FileManager Toolbar. The Toolbar includes some built-in and custom tools.
+
+````CSHTML
`````
Make sure that all comments that are complete sentences start with a capital letter and have the corresponding end punctuation.

ntacheva commented:
I've addressed the suggestions and refactored the article to follow the [PDFViewer Model](https://docs.telerik.com/blazor-ui/components/pdfviewer/toolbar)

yordan-mitev commented on components/filemanager/toolbar.md:
```diff
@@ -10,36 +10,243 @@ position: 15
+
+| Tool Name | Tool Tag | Description |
+| --- | --- | --- |
+| New Folder | `FileManagerToolBarNewFolderTool` | A button that creates a new folder in the current location. Read how to handle the creation at [FileManager Events]({%slug filemanager-events%}). |
```
```suggestion
| New Folder | `FileManagerToolBarNewFolderTool` | A button that creates a new folder in the current location. Read how to handle the creation in the [FileManager Events]({%slug filemanager-events%}) article.|
```

yordan-mitev commented on components/filemanager/toolbar.md:
```diff
@@ -10,36 +10,243 @@ position: 15
+| --- | --- | --- |
+| New Folder | `FileManagerToolBarNewFolderTool` | A button that creates a new folder in the current location. Read how to handle the creation at [FileManager Events]({%slug filemanager-events%}). |
+| Upload | `FileManagerToolBarUploadTool` | A button that opens a dialog with integrated [Upload component]({%slug upload-overview%}). Read more in the [FileManager Upload]({%slug filemanager-upload%}) article. |
+| Sort Direction | `FileManagerToolBarSortDirectionTool` | A ButtonGroup with ToggleButtons that selects the [sort direction(ascending or descending) to sort the files in the current location]({%slug filemanager-sort%}). |
```
```suggestion
| Sort Direction | `FileManagerToolBarSortDirectionTool` | A ButtonGroup with ToggleButtons that selects the [sort direction (ascending or descending) to sort the files in the current location]({%slug filemanager-sort%}). |
```

yordan-mitev commented on components/filemanager/toolbar.md:
```diff
@@ -10,36 +10,243 @@ position: 15
+| Sort Direction | `FileManagerToolBarSortDirectionTool` | A ButtonGroup with ToggleButtons that selects the [sort direction(ascending or descending) to sort the files in the current location]({%slug filemanager-sort%}). |
+| Sort By | `FileManagerToolBarSortTool` | A SplitButton that selects the desired [sort member to sort by]({%slug filemanager-sort%}). |
+| Views | `FileManagerToolBarFileViewTool` | A ButtonGroup with ToggleButtons that [toggles the file and folder visualization between a ListView and a Grid]({%slug filemanager-views%}). |
+| View Details | `FileManagerToolBarViewDetailsTool` | A Switch that toggles the visibility of the [FileManager Preview Pane]({%slug filemanager-preview-pane%}). By default the preview pane is hidden. |
```
```suggestion
| View Details | `FileManagerToolBarViewDetailsTool` | A Switch that toggles the visibility of the [FileManager Preview Pane]({%slug filemanager-preview-pane%}). By default, the preview pane is hidden. |
```

yordan-mitev commented on components/filemanager/toolbar.md:
```diff
@@ -10,36 +10,243 @@ position: 15
+
+By default, the FileManager Toolbar also includes a spacer (`<FileManagerToolBarSpacer />`). This spacer consumes the available empty space and pushes the rest of the tools next to one another.
+
+## Custom tools
```
```suggestion
## Custom Tools
```
